htmd/smallmol/chemlab/periodictable.py

```diff
@@ -0,0 +1,189 @@
+                           'DOUBLE': BondType.DOUBLE,
+                           'TRIPLE': BondType.TRIPLE}
+
+_atoms = {'H': [ 'H', 0, 0, '', 1, 1],
+         'C': ['C', 0, 0, '', 4, 1],
+        }
+
+class PeriodicTable:
+    """
+    Class that manages all the chemical information about atoms based on the atom. It exposes the rdkit PeriodicTable
+    functionalities.
+    """
+
+    def __init__(self):
+        self.PeriodicaTable = GetPeriodicTable()
+
+    def _evaluateMissingAtoms(self, valence, btypes):
+        """
+        Returns the difference bewtween the current valence computed and the expected one.
+
+        Parameters
+        ----------
+        valence: int
+            The valence of the atom
+        btypes: list
+            A list of the bondtypes connected to the atom
+
+        Returns
+        -------
+        missatoms: int
+            The number of atoms that miss for completing the atom valence
+
+        """
+        normbondvalence = sum([ 1.5 if int(b) == 12 else int(b) for b in btypes  ])
+
+        missatoms = int(valence - normbondvalence)
+
+        return missatoms
+
+    def getRadiusBond(self, element):
+        """
+        Returns the atom radius when bonded
+
+        Parameters
+        ----------
+        element: str
+            The element of the atom
+
+        Returns
+        -------
+        bradius: float
+            The radius of the bond when bonded
+
+        """
+
+        pT = self.PeriodicaTable
+        bradius = pT.GetRb0(element)
+
+        return bradius
+
+
+    def getMissingValence(self, atomidx, smallmol, formalcharge=0):
+        """
+        Returns the number of missing atom to complete the atom valence based on the types of bonds and formalcharge
+
+        Parameters
+        ----------
+        atomidx: int
+            The index of the atom
+        smallmol: htmd.smallmol.smallmol.SmallMol
+            The SmallMol object
+        formalcharge: int
+            The formalcharge of the atom
+            Default: 0
+
+        Returns
+        -------
+        missingatoms: int
+            The number of atom that are missing for completing the valence
+        """
+
+
+        pT = self.PeriodicaTable
+
+        element = smallmol.element[atomidx]
+        btypes = smallmol.bondtypes[atomidx]
+        #steric_number = len(smallmol.neighbors[atomidx])
+
+        valences = list(pT.GetValenceList(element))
+
+        # Some atoms can have multiple valid valences (like S, P). For them I should predict the most reliable one.
+        # At the moment not implemented. I will raise an error now.
+        if len(valences) == 1:
+            valence = valences[0]
+        else:
+            class Notmplemented(Exception):
+                pass
+            raise Notmplemented('The missing valence is not implemented for atom with more than one valid valence')
+
+        missingatoms = self._evaluateMissingAtoms(valence,  btypes)
```

Comment:
evaluateMissingElectrons no?

Reply:
Exactly as before. I know the valence and checking how many atoms are present at the moment I get how many are missing